components/NavigationBar/examples/NavigationBarIconsExample.m

```diff
@@ -77,10 +77,13 @@ - (void)viewDidLoad {
   self.navigationItem.rightBarButtonItem = trailingButtonItem;
   self.navigationItem.backBarButtonItem = backButtonItem;
 
-  if ([self.view respondsToSelector:@selector(safeAreaLayoutGuide)]) {
-    UILayoutGuide *layoutGuide = [self.view performSelector:@selector(safeAreaLayoutGuide)];
+  // TODO(rsmoore): Remove this check when we drop support for Xcode 7/8
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
```

Comment:
If this is gonna be used a lot, can we get more reusable code like a macro macro?

Reply:
Positive: We could simplify this and eliminate for the first __IPHONE_11_0 to get out of sync with the second.
Negative: Why verbose this is 'standard' macro magic that most iOS engineers can immediate parse / recognize.
If we do create a macro it should likely be standard across all of MDC-iOS and used universally.

Reply:
^ That was my hope (the second)

Reply:
How can we drop the first check (`defined (__IPHONE_11_0)`)? Won't that result in an using an undefined symbol in the second check?

Reply:
Sorry, I wasn't particularly clear.
Right now we are checking against a single value (__IPHONE_11_0) twice.
We could find ourselves in a position where we have the following error:
# if defined(__IPHONE_11_0)  && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_99_3)
Although in reality we end up with a dropped character or missing minor version.
By using a macro, we can take a single parameter to check against.
That said, I'm hesitant to create our own macro to add a layer of indirection over well-known pattern.

Reply:
That makes sense.  I think we should keep macros in all the places they are needed. I dislike them in general, but they bring attention to areas of code that are particularly gritty (like this example) and deserve attention.